fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java

```diff
@@ -91,7 +91,8 @@ public Boolean visitCaseWhenOperator(CaseWhenOperator operator, Void context) {
                 return false;
             }
             ScalarOperator thenClause = operator.getThenClause(i);
-            if (!Utils.extractColumnRef(thenClause).stream().allMatch(aggregateColumns::contains)) {
+            if (!Utils.extractColumnRef(thenClause).stream().allMatch(aggregateColumns::contains) ||
+                    (thenClause.isConstantRef() && !((ConstantOperator) thenClause).isNull())) {
```

Comment:
I think may you need add thenClause.isConstant() check, like
select k1, sum(case k1 when 1 then abs(1) end )...

Reply:
@ABingHuang

Reply:
`!((ConstantOperator) thenClause).isNull())` already include `thenClause.isConstant()`？ @Seaven

Reply:
no, `ConstantOperator` only a literal, like '1'/'a'/NULL, `isConstant` means maybe an expression, like `abs(1)`, `1 + 1`, `case 1=abs(2) then 2`, all his inputs are `ConstantOperator`, but himselft not a `ConstantOperator`. we can compute some constant expression result in optimizer, but not all

Reply:
thx for clear explanation，I have added henClause.isConstant() check. @Seaven